kernel/typing.ml

```diff
@@ -152,6 +152,56 @@ let unshift_reduce sg q t =
     ( try Some (Subst.unshift q (R.snf sg t))
       with Subst.UnshiftExn -> None )
 
+exception VarSurelyOccurs
```

Comment:
I think that the name of this exception is bad and does not convey the correct meaning

Reply:
I'm open suggestions

Reply:
By the way why this function is not local to the function where it is used as
```let foo ... =
    let exception VarSurelyOccurs in
    ...
``` ?
It would be better I think and the name makes more sense inside the type checking function.

Reply:
Apparently this is not a feature of OCaml 4.03

Reply:
Maybe we should not support that version anyway, it is getting old and with `opam` it is really easy to install a better version of OCaml.

Reply:
It is also really easy to support 4.03. Here the exception declaration can simple be moved up two lines in the code.